### PR TITLE
Add default value for `info` argument of `ClassDescription`

### PR DESCRIPTION
**Pull request by emilk**
* Closes https://github.com/rerun-io/rerun/issues/2924

Two changes:
* You can now easily specify the `info` without having to also specify label and color, using `info=0`.
* The default info is one with `id=0`, matching 0.7.0 behavior and how `log_points` work

(This took me over an hour to figure out 😭)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3017) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3017)
- [Docs preview](https://rerun.io/preview/pr%3Aemilk%2Ffix-annotation-context-default/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aemilk%2Ffix-annotation-context-default/examples)

### CODE REVIEW

**Review by Wumpf — APPROVED** [review]
not acutally sure if the default value is a good idea. "Give a title to this class" becomes odd when one doesn't specify which class is meant - or am I missunderstanding the situation?

**Comment by emilk** [comment]
I don't know - `AnnotationInfo` is too complicated for my brain.

Where are you quoting from? I can't find "Give a title to this class" anywhere

**Comment by Wumpf** [comment]
Oh that's not a quote, that's more of my interpretation of what giving a setting an annotationinfo means

File: rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py [hunk 1]
```diff
@@ -29,7 +29,8 @@
 
 def classdescription_init(
     self: ClassDescription,
-    info: AnnotationInfoLike,
+    info: AnnotationInfoLike = 0,
```
Comment (jleibs) [inline rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py:32]: This still strikes me as a bit of a footgun, but it matches the pre-existing behavior.

**Comment by emilk** [comment]
I don't like the implicit default value either. I think I'll merge this and cherry-pick for 0.8.1 (since it fixes a subtle 0.8.0 API regression) and then make a new PR to make `info` mandatory, making the user code more explicit and intentionally breaking the API

**Review by jleibs — APPROVED** [review]
Probably worth adding a comment to the generated docstring (in `class_description.fbs`) that if no info is provided, the default is 0.

The consequences of this really are that a line such as:
```
rr.ClassDescription(keypoint_connections=[(0, 1), (1, 2), (2, 0), (2, 3), (0, 4), (1, 5)])
```

Implicitly says to annotate keypoint_connections for keypoints that have a class_id of 0, which matches the behavior of `log_points`, which *sets* the class_id to 0 if you log keypoints without logging a class-id.

File: rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py (hunk 1 above)
Comment (emilk) [reply rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py:32]: I'm gonna remove it in a follow-up-PR, but keep it here to not break API further in a patch release

Comment (jleibs) [reply rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py:32]: If you want to break it in the future, why not break it now? Is there a meaningful difference between breaking it in 0.8 and breaking it in 0.9?

Comment (emilk) [reply rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py:32]: It's more that I want to do the least intrusive change for 0.8.1.

If we require `info` to be explicit, should we also make it explicit for `log_points`? Why require explicitness in one but not the other?

Comment (jleibs) [reply rerun_py/rerun_sdk/rerun/_rerun2/datatypes/_overrides/annotation_context.py:32]: Fair enough.

**Pull request merged**